### DISCUSSION TITLE
Custom ID in Keystone 6

### DISCUSSION THREAD

**Discussion by ksemanur8** [keystonejs/keystone#8920] [Questions] [answered]
My team built a database with Keystone 6 and we would like to put custom values in the ID field of our tables. 
To give some more context, we have a table called Projects. This table has a `ProjectId` field, which is defined as such: 
```
projectId: text({
        validation: { isRequired: true },
        isIndexed: "unique",
      }),
```
This field has all the properties we would want from a primary key for the table, so we find the auto-generated ID field unnecessary, and would like to customize it to be the same as `projectId`.
Here's how it looks on the Admin UI (Item ID is the auto-generated value).
<img width="881" alt="Screenshot 2023-11-16 at 11 59 46 AM" src="https://github.com/keystonejs/keystone/assets/59751315/45540336-e75c-4a0b-8133-cf84f1fbae25">

Based on [some discussions about this topic I found from Keystone 5 days](https://github.com/keystonejs/keystone/issues/1834#issuecomment-547211757), I've attempted setting an id field in our schema using the following syntax 
```
fields: {
   id: ....
}
```
However, I received this error:
<img width="840" alt="Screenshot 2023-11-16 at 12 07 07 PM" src="https://github.com/keystonejs/keystone/assets/59751315/e2e194d6-44cc-4e53-9198-26742aaa4538">

I also tried the `idField` option but found that it just makes a new field instead of controlling values in the pre-existing id field.

Is there another way of customizing the ID field?

**Comment by dcousens** [comment, marked as answer]
Hi @ksemanur8,

Did you review the [Custom ID](https://github.com/keystonejs/keystone/tree/main/examples/custom-id) example?
Although I conceptually agree with your intuition that `id` should be listed under `fields`, the current versions of `@keystone-6/core` don't support that.
